### PR TITLE
156005738 GTFS viewer

### PR DESCRIPTION
**Pull request by tatut**
# Added
- Preliminary GTFS view feature

### CODE REVIEW

**Comment by tatut** [comment]
Try it with: `#/routes/view-gtfs?url=<url to gtfs file>`

File: ote/src/cljs/ote/views/gtfs_viewer.cljs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    {:name :headsign}
+    {:name :departures
+     :read #(str/join ", " (:departures %))}]
+   (:trips selected-route)])
+
+(defn gtfs-viewer [e! {gtfs :gtfs-viewer}]
+  [:div.gtfs-viewer
+   [routes-table e! gtfs]
+   [trips-map e! gtfs]
+   #_[trips-table e! gtfs]])
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/gtfs_viewer.cljs]: commented code and the trips-table is also in comments.

File: ote/src/cljs/ote/app/routes.cljs [hunk 2]
```diff
@@ -24,9 +26,14 @@
     ["/route/new" :new-route]
     ["/edit-route/:id" :edit-route]
 
+    ["/routes/view-gtfs" :view-gtfs]
```
Comment (rckopteri) [inline ote/src/cljs/ote/app/routes.cljs]: I think that you could add this route to admin panel also (maybe not in this pr but soon) ?

File: ote/src/cljc/ote/util/zip.cljc [hunk 3]
```diff
@@ -23,3 +22,20 @@
               (.putNextEntry out (ZipEntry. name))
               (.write out (.getBytes data "UTF-8"))
               (.closeEntry out)))))
+
+#?(:clj
+   (defn read-zip
+     "Read a zip file. Returns a sequence of file descriptors.
+  Each file descriptor is a map containing the name and data."
+     [input]
+     (println "read from " input)
```
Comment (rckopteri) [inline ote/src/cljc/ote/util/zip.cljc]: Could change to (log/info )

**Pull request merged**